python/pyspark/shuffle.py

```diff
@@ -512,9 +512,6 @@ def load(f):
                     f.close()
                 chunks.append(load(open(path, 'rb')))
                 current_chunk = []
-                gc.collect()
```

Comment:
We won't change `limit` to `_next_limit()` (which call get_used_memory()). This line here was to get better number about how much memory was used, is not needed anymore.